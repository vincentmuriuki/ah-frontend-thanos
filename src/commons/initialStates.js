const initialState = {
  articleReducer: {
    article: {
      id: 0,
      tag_list: [],
      slug: '',
      title: '',
      description: '',
      body: '',
      created_at: '',
      updated_at: '',
      image_url: '',
      audio_url: '',
      read_time: '',
      likes: 0,
      dislikes: 0,
      likeDislikeStatus: '',
      views_count: '',
      rating: 0,
      author: {
        id: 0,
        username: '',
        email: '',
      },
    },
    articles: [],
    errorMessage: '',
  },
  deleteArticleReducer: {},
  socialLoginReducer: { isLoggedIn: false },
  loginReducer: { errorMessage: '', successMessage: '', user_details: '' },
  userReducer: {
    freshUser: { email: '', password: '', username: '' },
  },
};

export const sampleArticle = {
  id: 1,
  tag_list: ['tag'],
  slug: 'title-sample',
  title: 'title sample',
  description: 'desc',
  body: 'body',
  created_at: '12-12-2018',
  updated_at: '12-12-2018',
  image_url: 'http://test.png',
  audio_url: 'http://test.mp3',
  read_time: '0 min',
  likes: 12,
  dislikes: 2,
  likeDislikeStatus: 'dislike',
  views_count: 0,
  rating: 0,
  author: {
    id: 2,
    username: 'janedoe',
    email: 'janedoe@gmail.com',
  },
};

export const sampleArticle2 = {
  ...sampleArticle,
  likeDislikeStatus: 'like',
};

export const initialStateWithSample = {
  article: sampleArticle,
};

export const initialStateWithSample2 = {
  article: sampleArticle2,
};

export default initialState;
