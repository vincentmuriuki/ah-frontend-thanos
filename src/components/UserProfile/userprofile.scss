$primaryLight: #40e0d0;
$primaryDark: #20c0b0;
$primaryDarker: #345F5B;

.profile-box {
  display: flex;
  flex-flow: row wrap;
  padding: 30px 0;

  .grey {
    color: #888888;
  }

  .icon path {
    fill: turquoise;
  }

  .lefter, .righter {
    flex-grow: 1;
    min-width: 200px;
    padding: 15px;
    border: 1px solid #eeeeee;
  }

  .lefter {
    width: 25%;
    text-align: center;

    .profile-image {
      width: 250px;
      height: 250px;
      overflow: hidden;
      border: 1px solid turquoise;
        
      img {
        width: 250%;
        margin-top: -50px;
        margin-left: -50px;
        border-radius: 1px solid #eeeeee;
      }
    }
    .profile-username {
      padding: 10px 0;
    }
  }
  .righter {
    width: 75%;

    .profile-first-name, .profile-last-name, .profile-bio, .profile-created-at {
      padding: 5px 0;
    }
  }
}

.follow-box {

  .followers-box, .followees-box {
    border: 1px solid #eaeaea;
    padding: 10px;
    display: flex;
    flex-flow: row wrap;
    
    h6 {
     width: 100%;
     color: $primaryDarker;
    }
    .follower-image {
      width: 50px;
      height: 50px;
      overflow: hidden;
      border-radius: 50%;
      margin: 10px;
  
      img {
        width: 100%;
      }
    }
    .follower-image:hover {
      border: 2px solid $primaryDark;
    }
  }
  .followees-box {
    margin-bottom: 40px;
  }
}
