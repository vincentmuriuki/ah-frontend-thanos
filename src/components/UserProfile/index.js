// show profile of any registered user (given the username in the URL)
import React from 'react';
import PropTypes from 'prop-types';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUser } from '@fortawesome/free-solid-svg-icons';
import Follow from '../Follow';
import './userprofile.scss';

library.add(faUser);

const showFollowerFolloweeProfiles = (image, username, firstName, lastName) => (
  <div key={username}>
    <div className="follower-image">
      <a href={`/profiles/${username}`}>
        {image
          ? <img src={image} title={`${firstName} ${lastName} (${username})`} alt="dp" />
          : <img src="https://iviidev.info/downloads/images/anon.png" alt="dp" />}
      </a>
    </div>
  </div>
);

const createFollowDiv = (cssClass, title, list) => (
  <div className={cssClass}>
    <h6>{title}</h6>
    {list.map(
      user => showFollowerFolloweeProfiles(
        user.image, user.username, user.first_name, user.last_name,
      ),
    )}
  </div>
);

const UserProfile = ({
  currentProfile, onClick, followersList, followeesList,
}) => (
  <div className="container">
    <div className="profile-box">
      <div className="lefter">
        <div className="profile-image">
          <img src={currentProfile.image} alt="profile dp" />
        </div>
        <div className="profile-username">
          <FontAwesomeIcon className="icon" icon="user" />
          <span className="ml-2">{currentProfile.username}</span>
        </div>
        <Follow onClick={onClick} option={currentProfile.isFollowee} />
      </div>
      <div className="righter">
        <div className="profile-first-name">
          <div className="small grey">First Name:</div>
          <span>{currentProfile.first_name}</span>
        </div>
        <div className="profile-last-name">
          <div className="small grey">Last Name:</div>
          <span>{currentProfile.last_name}</span>
        </div>
        <div className="profile-bio">
          <div className="small grey">Bio:</div>
          <span>{currentProfile.bio}</span>
        </div>
        <div className="profile-created-at">
          <div className="small grey">Proud user since:</div>
          <span>{currentProfile.created_at}</span>
        </div>
      </div>
    </div>
    <div className="follow-box">
      {createFollowDiv('followers-box', `People following ${currentProfile.username}`, followersList)}
      {createFollowDiv('followees-box', `People ${currentProfile.username} follows`, followeesList)}
    </div>
  </div>
);

UserProfile.propTypes = {
  currentProfile: PropTypes.shape({}).isRequired,
  onClick: PropTypes.func.isRequired,
  followersList: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  followeesList: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default UserProfile;
