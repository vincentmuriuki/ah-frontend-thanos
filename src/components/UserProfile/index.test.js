import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from '.';
import initialStates from '../../commons/initialStates';

describe('<UserProfile /> component', () => {
  let wrapper;

  beforeEach((() => {
    const { currentProfile } = initialStates.followUnfollowReducer;
    wrapper = shallow(
      <UserProfile
        currentProfile={currentProfile}
        onClick={jest.fn()}
        followersList={[currentProfile]}
        followeesList={[
          { ...currentProfile, image: 'https://picsum.photos/600' },
        ]}
      />,
    );
  }));

  it('renders UserProfile', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
