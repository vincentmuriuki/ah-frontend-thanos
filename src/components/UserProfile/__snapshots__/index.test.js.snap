// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> component renders UserProfile 1`] = `
<div
  className="container"
>
  <div
    className="profile-box"
  >
    <div
      className="lefter"
    >
      <div
        className="profile-image"
      >
        <img
          alt="profile dp"
          src=""
        />
      </div>
      <div
        className="profile-username"
      >
        <FontAwesomeIcon
          border={false}
          className="icon"
          fixedWidth={false}
          flip={null}
          icon="user"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span
          className="ml-2"
        />
      </div>
      <Follow
        onClick={[MockFunction]}
        option={false}
      />
    </div>
    <div
      className="righter"
    >
      <div
        className="profile-first-name"
      >
        <div
          className="small grey"
        >
          First Name:
        </div>
        <span />
      </div>
      <div
        className="profile-last-name"
      >
        <div
          className="small grey"
        >
          Last Name:
        </div>
        <span />
      </div>
      <div
        className="profile-bio"
      >
        <div
          className="small grey"
        >
          Bio:
        </div>
        <span />
      </div>
      <div
        className="profile-created-at"
      >
        <div
          className="small grey"
        >
          Proud user since:
        </div>
        <span />
      </div>
    </div>
  </div>
  <div
    className="follow-box"
  >
    <div
      className="followers-box"
    >
      <h6>
        People following 
      </h6>
      <div
        key=""
      >
        <div
          className="follower-image"
        >
          <a
            href="/profiles/"
          >
            <img
              alt="dp"
              src="https://iviidev.info/downloads/images/anon.png"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="followees-box"
    >
      <h6>
        People  follows
      </h6>
      <div
        key=""
      >
        <div
          className="follower-image"
        >
          <a
            href="/profiles/"
          >
            <img
              alt="dp"
              src="https://picsum.photos/600"
              title="  ()"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
